var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -3,185 +3,122 @@
-        libtool_inc = '-I %s/share/aclocal/' % (self.spec['libtool'].prefix)
-        newpath = 'aclocal -I m4 %s %s' % (pkgconf_inc, libtool_inc)
-        filter_file(r'aclocal -I m4', r'%s' % newpath, "build.sh")
-
-    def get_arch_options(self, spec):
-        options = []
-
-        if spec.satisfies('+cross-compile'):
-            options.extend(['cross_compiling=yes',
-                            '--without-memacs',
-                            '--without-nmodl'])
-
-        # on os-x disable building carbon 'click' utility
-        if 'darwin' in self.spec.architecture:
-            options.append('macdarwin=no')
-
-        return options
-
-    def get_python_options(self, spec):
-        options = []
-
-        if spec.satisfies('+python'):
-            python_exec = spec['python'].command.path
-            py_inc = spec['python'].headers.directories[0]
-            py_lib = spec['python'].prefix.lib
-
-            if not os.path.isdir(py_lib):
-                py_lib = spec['python'].prefix.lib64
-
-            options.extend(['--with-nrnpython=%s' % python_exec,
-                            '--disable-pysetup',
-                            'PYINCDIR=%s' % py_inc,
-                            'PYLIBDIR=%s' % py_lib])
-
-            if spec.satisfies('~cross-compile'):
-                options.append('PYTHON_BLD=%s' % python_exec)
-
+    url      = "https://neuron.yale.edu/ftp/neuron/versions/v7.7/nrn-7.7.tar.gz"
+    git      = "https://github.com/neuronsimulator/nrn"
+    maintainers = ['pramodk', 'nrnhines', 'iomaganaris', 'alexsavulescu']
+
+    version('develop', branch='master', submodules='True')
+    version("7.8.2", tag="7.8.2", submodules='True')
+    version("7.8.1", tag="7.8.1", submodules='True')
+
+    variant("coreneuron",    default=False, description="Enable CoreNEURON as submodule")
+    variant("cross-compile", default=False, description="Build for cross-compile environment")
+    variant("interviews",    default=False, description="Enable GUI with INTERVIEWS")
+    variant("legacy-unit",   default=False, description="Enable legacy units")
+    variant("mpi",           default=True,  description="Enable MPI parallelism")
+    variant("python",        default=True,  description="Enable python")
+    variant("rx3d",          default=False,  description="Enable cython translated 3-d rxd")
+    variant("tests",         default=False, description="Enable unit tests")
+
+    depends_on("bison",     type="build")
+    depends_on("flex",      type="build")
+    depends_on("py-cython", when="+rx3d", type="build")
+
+    depends_on("gettext")
+    depends_on("mpi",         when="+mpi")
+    depends_on("ncurses")
+    depends_on("python@2.7:", when="+python")
+    depends_on("py-pytest",   when="+python+tests")
+    depends_on("readline")
+
+    conflicts("+rx3d",        when="~python")
+
+    patch("patch-v782-git-cmake-avx512.patch", when="@7.8.2")
```

Comment:
Is the patch available as a commit / PR somewhere?

Reply:
Yes, I am preparing that in upstream https://github.com/neuronsimulator/nrn/pull/902/files